benchmark/benchmark.py

```diff
@@ -85,24 +90,48 @@ def get_numpy_benchmarks():
         name = filename.stem
         if name in SKIP:
             continue
-        content = parse_numpy_benchmark(filename)
-        content += (
-            "import numpy as np\n"
-            "_ = np.empty(())\n"
-            "setup = setup + '\\nfrom __main__ import {}'\n"
-            "from timeit import Timer\n"
-            "t = Timer(run, setup)\n"
-            "r = t.repeat(11, 40)\n"
-            "r.remove(min(r))\n"
-            "r.remove(max(r))\n"
-            "print(np.mean(r))\n".format(name)
-        )
-        yield name, content
+        if "canvas" not in str(filename) and "wasm" not in str(filename):
+            content = parse_numpy_benchmark(filename)
+            content += (
+                "import numpy as np\n"
+                "_ = np.empty(())\n"
+                "setup = setup + '\\nfrom __main__ import {}'\n"
+                "from timeit import Timer\n"
+                "t = Timer(run, setup)\n"
+                "r = t.repeat(11, 40)\n"
+                "r.remove(min(r))\n"
+                "r.remove(max(r))\n"
+                "print(np.mean(r))\n".format(name)
+            )
+            yield name, content
+
+
+def get_matplotlib_benchmarks():
+    root = Path(__file__).resolve().parent / "benchmarks"
+    for filename in sorted(root.iterdir()):
+        name = filename.stem
+        if name in SKIP:
+            continue
+        if "canvas" in str(filename) or "wasm" in str(filename):
+            content = parse_numpy_benchmark(filename)
+            content += (
+                "import numpy as np\n"
+                "_ = np.empty(())\n"
+                "setup = setup + '\\nfrom __main__ import {}'\n"
+                "from timeit import Timer\n"
+                "t = Timer(run, setup)\n"
+                "r = t.repeat(11, 20)\n"
```

Comment:
Related to the run time, for matplotlib you could try to do,
```suggestion
"r = t.repeat(repeat=3, number=1)"
```
a large `number` is for benchmarks that take ms and where the timer accuracy is going to make results less accurate. If plotting takes second, there is no reason to make several calls in a loop. Though you still need a couple of `repeat` to get statistics.

And maybe with this, we can keep all benchmarks, if it only adds a couple of minutes to the runtime.

Reply:
Let's do this in separate PR too. We could discuss which would be better:

- Running multiple benchmarks (numpy, pystone, matplotlib) in separate jobs
- Adjusting `repeat` and `number` of benchmark